### PR TITLE
Append tag name to the default image

### PR DESCRIPTION
**Pull request by vpavlin**
This PR fixes an issue where default image did not have a tag attached and OpenShift automatically appended `latest` tag, which is not correct.

It also moves the `default` annotation to the tag level to make sure we can make sure a specific tag is used as default

### CODE REVIEW

**Review by jeff-phillips-18 — APPROVED** [review]
LGTM, we'll need a follow up to fix the UI to use the `default` flag on the image tag.

**Comment by jeff-phillips-18** [comment]
@vpavlin Is there a difference between `recommended` and `default` on tags?

**Comment by vpavlin** [comment]
That's an interesting point...the main differenc in my mind is that `default` is set only once and it is the image used when a user does not select anything.

Recommended should be set in one tag for each image to make sure UI can select the right tag when user only picks an image.

Do you want me to add comments for this?

**Comment by jeff-phillips-18** [comment]
> Do you want me to add comments for this?

Up to you, I just need to know which tag the UI should select by default (I assume recommended if there is one, otherwise default, otherwise latest version).

File: jupyterhub_singleuser_profiles/images.py [hunk 1]
```diff
@@ -111,15 +115,15 @@ def load(self):
                                             ),
                                             name=tag.name,
                                             recommended=tag_annotations.get(RECOMMENDED_ANNOTATION, False),
-                                            build_status=self.get_image_build_status(imagestream_name)
+                                            build_status=self.get_image_build_status(imagestream_name),
+                                            default=bool(strtobool(annotations.get(DEFAULT_IMAGE_ANNOTATION, "False")))
```
Comment (mroman-redhat) [inline jupyterhub_singleuser_profiles/images.py:119]: If this has been moved to tag information, maybe it should use tag_annotations instead.

Comment (vpavlin) [reply jupyterhub_singleuser_profiles/images.py]: Good catch!

**Pull request merged**